internal/api/moduleloaderdata.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ModuleLoaderData struct {
```

Comment:
Is there a way to somehow generate that object based on the API? So we can regenerate on each API change?

Reply:
not that i know of right now. Mergo can probably help, but this is for later investigation

Reply:
Should this type also have a `KernelVersion` field?

Reply:
If this is a type that we will be passing around everywhere, should we make it an interface? We could change the implementation as we please.

Reply:
1) comment added
2) kernelVersion added
3) i don't see the benefit: this struct just replaces the KernelMapping and the Module structures. Adding it as interface will just make code more complicated. Currently the only change that i foresee is adding a new field to the structure. If we rename the variable, then fixing it will be a simple matter of recompiling